libs/components/datetime/src/lib/modules/datepicker/calendar/daypicker-cell.component.ts

```diff
@@ -28,141 +29,94 @@ import { SkyDayPickerContext } from './daypicker-context';
-  }
+  protected hasTooltip = computed(() => {
+    const date = this.date();
 
-  public ngOnInit(): void {
-    this.hasTooltip = !!(
-      this.date &&
-      this.date.keyDate &&
-      this.date.keyDateText &&
-      this.date.keyDateText.length > 0 &&
-      this.date.keyDateText[0].length > 0
+    return (
+      date &&
+      date.keyDate &&
+      date.keyDateText &&
+      date.keyDateText.length > 0 &&
+      date.keyDateText[0].length > 0
     );
+  });
 
+  protected popoverController = new Subject<SkyPopoverMessage>();
+
+  constructor() {
     // show the tooltip if this is the active date and is not the
     // initial active date (activeDateHasChanged)
-    if (
-      !!this.date &&
-      this.#datepicker.isActive(this.date) &&
-      this.activeDateHasChanged &&
-      this.hasTooltip
-    ) {
-      this.#activeUid = this.date.uid;
-      this.#showTooltip();
-    }
-
-    if (this.hasTooltip) {
-      this.#datepickerService.keyDatePopoverStream
-        .pipe(takeUntil(this.#ngUnsubscribe))
-        .subscribe((date) => {
-          if (date) {
-            this.#activeUid = date.uid;
-          } else {
-            this.#activeUid = '';
-          }
-          // If this day has an open popover and they have moved off of the day close the popover.
-          if (this.date?.uid !== this.#activeUid) {
-            this.#hideTooltip();
-          }
-        });
-    }
-  }
-
-  public ngOnDestroy(): void {
-    this.#ngUnsubscribe.next();
-    this.#ngUnsubscribe.complete();
+    effect(() => {
+      const activeDateHasChanged = this.activeDateHasChanged();
+      const hasTooltip = this.hasTooltip();
+      const date = this.date();
+
+      if (
+        date &&
+        activeDateHasChanged &&
+        hasTooltip &&
+        this.#datepicker.isActive(date)
+      ) {
+        this.#showTooltip();
+      }
+    });
+
+    // Hide this cell's tooltip if another cell opens a tooltip.
+    this.#calendarSvc.keyDatePopoverStream
+      .pipe(takeUntilDestroyed())
+      .subscribe((popoverDate) => {
+        if (!popoverDate || popoverDate?.uid !== this.date()?.uid) {
+          this.#hideTooltip();
+        }
+      });
   }
 
-  public onDayMouseenter(): void {
-    this.#cancelPopover = false;
-    if (this.hasTooltip) {
+  protected onDayMouseenter(): void {
+    if (this.hasTooltip()) {
       this.#showTooltip();
-      this.#datepickerService.keyDatePopoverStream.next(this.date);
+      this.#calendarSvc.keyDatePopoverStream.next(this.date());
     }
   }
 
-  public onDayMouseleave(): void {
-    this.#cancelPopover = true;
-    if (this.hasTooltip) {
+  protected onDayMouseleave(): void {
+    if (this.hasTooltip()) {
       this.#hideTooltip();
+      this.#calendarSvc.keyDatePopoverStream.next(undefined);
     }
-    this.#datepickerService.keyDatePopoverStream.next(undefined);
-  }
-
-  public onPopoverClosed(): void {
-    this.#popoverOpen = false;
-  }
-
-  public onPopoverOpened(): void {
```

Comment:
All of this logic was overcomplicated and created timing issues. It can be dramatically simplified with signals.